### ISSUE TITLE
LOC-22 ⁃ Trouble with provisioning Lambda

### ISSUE THREAD

**Issue by davekiss** [localstack/aws-cdk-local#4]
Hey, awesome project, can't wait to get it working

I have a simple cdk template for a Lambda stack based off of the TestStack provided by `aws-cdk-local`:

<details>
<summary>Template</summary>

```
const cdk = require("@aws-cdk/core");
const lambda = require("@aws-cdk/aws-lambda");
const path = require("path");

class LambdaStack extends cdk.Stack {
  /**
   * @param {cdk.App} scope
   * @param {string} id
   * @param {cdk.StackProps=} props
   */
  constructor(scope, id, props) {
    super(scope, id, props);

    new lambda.Function(this, "MyCdkFunction", {
      runtime: lambda.Runtime.NODEJS_12_X,
      handler: "index.handler",
      code: lambda.Code.fromAsset(
        path.join(__dirname, "./functions/helloWorld")
      )
    });
  }
}

module.exports = { LambdaStack };
```

</details>

I ran `cdklocal bootstrap` as a requirement of creating an S3 bucket to upload the asset to.

Then, I ran `cdklocal deploy` and all seems well:

<details>
<summary>Deploy output</summary>

```
(node:42109) ExperimentalWarning: The fs.promises API is experimental
This deployment will make potentially sensitive changes according to your current security approval level (--require-approval broadening).
Please confirm you intend to make the following modifications:

IAM Statement Changes
┌───┬──────────────────────────────────┬────────┬────────────────┬──────────────────────────────┬───────────┐
│   │ Resource                         │ Effect │ Action         │ Principal                    │ Condition │
├───┼──────────────────────────────────┼────────┼────────────────┼──────────────────────────────┼───────────┤
│ + │ ${MyCdkFunction/ServiceRole.Arn} │ Allow  │ sts:AssumeRole │ Service:lambda.amazonaws.com │           │
└───┴──────────────────────────────────┴────────┴────────────────┴──────────────────────────────┴───────────┘
IAM Policy Changes
┌───┬──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────┐
│   │ Resource                     │ Managed Policy ARN                                                             │
├───┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────┤
│ + │ ${MyCdkFunction/ServiceRole} │ arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole │
└───┴──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────┘
(NOTE: There may be security-related changes not in this list. See https://github.com/aws/aws-cdk/issues/1299)

Do you wish to deploy these changes (y/n)? y
LambdaStack: deploying...
[0%] start: Publishing 6f8229de3e2b9365627caceef4302866d709139f4656d8b5e887e8e48e83e362:current
[100%] success: Published 6f8229de3e2b9365627caceef4302866d709139f4656d8b5e887e8e48e83e362:current
LambdaStack: creating CloudFormation changeset...
[··························································] (0/4)


 ✅  LambdaStack

Stack ARN:
arn:aws:cloudformation:us-east-1:000000000000:stack/LambdaStack/fb30a692-bb5a-4a29-b474-3f1c83e8def2
```

</details>

The lambda stack appears in my localstack cloud dashboard

![image](https://user-images.githubusercontent.com/1256071/97723670-6f581600-1aa2-11eb-92c5-c7d95c493d58.png)

However, nothing appears in functions

![image](https://user-images.githubusercontent.com/1256071/97723774-8434a980-1aa2-11eb-8c58-889ea95766e0.png)

And the following returns no results:

```
$ awslocal lambda list-functions

{
    "Functions": []
}
```

I tried to make a change to the function and redeploy it, and got this message:

```
 LambdaStack failed: Missing Parameter: Missing parameter AssetParametersb5f1908c5949d9b30e17a8f5c68ce01e0938d296338da3108b88097945ef6395S3Bucket4CA074F4
```

I've attempted to delete my `cdk.out` folder, `destroy` and then `deploy` again, but no success.

Any advice is appreciated! Let me know if I can provide more information.

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-22) by [Unito](https://www.unito.io/learn-more)

**Comment by ekbarber** [comment]
I'm running into a similar issue. The first deploy of a Stack with a lambda using local code (i.e. `code: Code.fromAsset('./dist')` in it works perfectly fine. However, any subsequent deploys after changing the code locally result in the error mentioned above:
```
[STACK_NAME] failed: Missing Parameter: Missing Parameter AssetParameterse975810db87d3a638be2160e424ee4700bb8104d7850eeda04f463b22eb3a4e5S3Bucket91A09BE2
```

Additionally, the DEBUG log from localstack shows the following when this occurs:
```
2020-11-10T15:20:01:DEBUG:localstack.services.cloudformation.cloudformation_listener: Error response for CloudFormation action "CreateChangeSet" (400) POST /: b'<ErrorResponse xmlns="http://cloudformation.amazonaws.com/doc/2010-05-15/">\n  <Error>\n    <Type>Sender</Type>\n    <Code>Missing Parameter</Code>\n    <Message>Missing parameter AssetParameterse975810db87d3a638be2160e424ee4700bb8104d7850eeda04f463b22eb3a4e5S3Bucket91A09BE2</Message>\n  </Error>\n  <RequestId>cf4c737e-5ae2-11e4-a7c9-ad44eEXAMPLE</RequestId>\n</ErrorResponse>'
```

**Comment by guoliang** [comment]
I have the same issue https://github.com/localstack/aws-cdk-local/issues/35

**Comment by whummer** [comment]
Thanks for reporting @davekiss @ekbarber @guoliang . This should be fixed in localstack/localstack#3307 - can you please pull the latest Docker image and give it another try? Please report here if the problem persists. Thanks
